packages/wonder-blocks-birthday-picker/src/components/birthday-picker.js

```diff
@@ -0,0 +1,259 @@
+ * dropdowns to achieve a similar effect.
+ *
+ * More information on this pattern:
+ * https://medium.com/samsung-internet-dev/making-input-type-date-complicated-a544fd27c45a
+ *
+ * Arguably, this should probably even be 3 textfields, but that would be a
+ * larger design change, more info:
+ * https://designnotes.blog.gov.uk/2013/12/05/asking-for-a-date-of-birth/
+ *
+ * **NOTE:** This component is uncontrolled.
+ *
+ * ### Usage
+ *
+ * ```jsx
+ * import {BirthdayPicker} from "@khanacademy/wonder-blocks-dates";
+ *
+ * <BirthdayPicker
+ *  defaultValue="2021-05-26"
+ *  onChange={(date) => {setDate(date)}}
+ * />
+ * ```
+ */
+export default class BirthdayPicker extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        this.lastChangeValue = props.defaultValue || null;
+        this.state = this.getStateFromDefault();
+    }
+
+    /**
+     * Calculates the initial state values based on the default value.
+     */
+    getStateFromDefault(): State {
+        const {defaultValue} = this.props;
+        const initialState: State = {
+            month: null,
+            day: null,
+            year: null,
+            error: null,
+        };
+
+        // If a default value was provided then we use moment to convert it
+        // into a date that we can use to populate the
+        if (defaultValue) {
+            const date = moment(defaultValue);
+
+            if (date.isValid()) {
+                initialState.month = String(date.month());
+                initialState.day = String(date.date());
+                initialState.year = String(date.year());
+            }
+
+            // If the date is in the future or is invalid then we want to show
+            // an error to the user.
+            if (date.isAfter() || !date.isValid()) {
+                initialState.error = BIRTHDAY_ERROR;
+            }
+        }
+
+        return initialState;
+    }
+
+    lastChangeValue: ?string = null;
+
+    /**
+     * Report changes back to the calling component, but only if the value
+     * has actually changed since the last time it was reported
+     * (or initialized).
+     *
+     * @param value the value to report back to the calling component.
+     */
+    reportChange: (value: ?string) => void = (value) => {
+        if (value !== this.lastChangeValue) {
+            this.lastChangeValue = value;
+            this.props.onChange(value);
+        }
+    };
+
+    /**
+     * Handle a change to any of the input fields, confirming if the input is
+     * valid, and then reporting the result back to the calling component via
+     * reportChange.
+     */
+    handleChange: () => void = (): void => {
+        const {month, day, year} = this.state;
+
+        // If any of the values haven't been set then our overall value is
+        // equal to null
+        if (month === null || day === null || year === null) {
+            this.reportChange(null);
+            return;
```

Comment:
I see the change here. I think GH pinned the comment to the commit somehow, or something.